### ISSUE TITLE
channels_test is flaky in SNP builds

### ISSUE THREAD

**Issue by achamayou** [microsoft/CCF#5853]
Failure observed in https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=79185&view=logs&jobId=4bdd83b0-b422-5e84-0058-fb00a4da194e&j=4bdd83b0-b422-5e84-0058-fb00a4da194e&t=01ab89b5-2bb2-5b36-1c41-59a63e14a55c

```
24: Test command: /CCF/build/channels_test
24: Environment variables: 
24:  TSAN_OPTIONS=suppressions=/CCF/tsan_env_suppressions
24:  ASAN_OPTIONS=alloc_dealloc_mismatch=0
24: Test timeout computed to be: 10000000
24: [doctest] doctest version is "2.4.11"
24: [doctest] run with "--help" for options
24: ===============================================================================
24: ../src/node/test/channels.cpp:1405:
24: TEST CASE:  Key rotation
24: 
24: ../src/node/test/channels.cpp:1562: FATAL ERROR: REQUIRE( received_by_2 == expected_received_by_2 ) is NOT correct!
24:   values: REQUIRE( {?} == {?} )
24: 
24: ===============================================================================
24: [doctest] test cases:   11 |   10 passed | 1 failed | 0 skipped
24: [doctest] assertions: 1517 | 1516 passed | 1 failed |
24: [doctest] Status: FAILURE!
24/57 Test #24: channels_test ....................***Failed    6.19 sec
```

**Comment by maxtropets** [comment]
https://github.com/microsoft/CCF/pull/6260

**Comment by maxtropets** [comment]
Follow-up fix: #6265
